instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/auto/instrumentation/opentelemetryapi/trace/UnshadedTracerProvider.java

```diff
@@ -42,4 +53,20 @@ public Tracer get(final String instrumentationName, final String instrumentation
             .get(instrumentationName, instrumentationVersion),
         contextStore);
   }
+
+  // this is called by OpenTelemetrySdk, which expects to get back a real TracerProviderSdk
+  @Override
+  public Object unobfuscate() {
+    if (!messageAlreadyLogged.getAndSet(true)) {
+      String message =
+          "direct usage of the OpenTelemetry SDK is not supported when running agent"
+              + " (run with debug logging to see stack trace)";
+      if (log.isDebugEnabled()) {
+        log.debug(message, new Exception("stack trace"));
+      } else {
+        log.info(message);
+      }
+    }
+    return ((Obfuscated<?>) originalTracerProvider).unobfuscate();
```

Comment:
Sorry, but I am totally confused. Javadoc for `Obfuscated` says: "because auto-instrumentation takes over the static global providers returned by the API and points them to it's embedded SDK". And also "Static global providers are obfuscated when they are returned from the API to prevent users from casting them to their SDK specific implementation."

But it seems to me that here we are doing exactly the same, no? We don't "take over the static global providers returned by the API and point them to it's embedded SDK", but we take global provider, unobfuskcate it and return to the end users.

Can we have a chat where @trask we explain to me (and @anuraaga if he wishes so) this whole business with obfuscation, shading, future support for multiple versions of API/SDK etc etc. Because I am totally confused.